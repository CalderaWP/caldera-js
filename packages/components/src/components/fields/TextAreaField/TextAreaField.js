import {
	parseAttributes,
	fieldClassNames,
	isValidHtml5type,
	toBoolean
} from '../util';

import PropTypes from 'prop-types';
import React from 'react';
import { TextareaControl } from '@wordpress/components';
export const TextAreaField = ({
	label,
	description,
	fieldId,
	placeholder,
	required,
	html5type,
	value,
	onChange,
	onBlur,
	attributes,
}) => {
	const fieldType = isValidHtml5type(html5type) ? html5type : 'text';
	const _attributes = parseAttributes(attributes, 'textarea');
	const rows = _attributes.hasOwnProperty('rows') ? _attributes.rows : 6;
	return(
		<TextareaControl
			rows={rows }
			label={label}
			id={fieldId}
			required={required}
			value={value}
			help={description}
			onChange={onChange}
			onBlur={onBlur}
			{..._attributes}
		/>
	)
};

TextAreaField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	attributes: PropTypes.object,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.array,
		PropTypes.bool
	]),
	onChange: PropTypes.func,
	onBlur: PropTypes.func
};

TextAreaField.defaultProps = {
	onBlur: () => {},
	required: false,
};
