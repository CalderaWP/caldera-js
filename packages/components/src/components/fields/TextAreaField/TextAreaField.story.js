import React from 'react';
import { storiesOf } from '@storybook/react';
import { TextAreaField } from './TextAreaField';

let value = 'Hi Roy';
const onChange = newValue => {
	console.log(newValue);
};
const onBlur = event => {};


storiesOf('TextAreaField', module).add('With no value', () => (
	<TextAreaField
		label={'Text Input'}
		onChange={onChange}
		onBlur={onBlur}
	/>
));

