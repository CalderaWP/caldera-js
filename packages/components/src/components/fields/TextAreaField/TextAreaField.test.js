import * as React from 'react';
import { mount } from 'enzyme';
import {cleanup, render} from "react-testing-library";
import { TextAreaField } from './TextAreaField';

describe('TextAreaField ', () => {
	let onChange;
	let onBlur;

	afterEach(cleanup);


	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('matches snapshot with all props', () => {
		const component = render(
			<TextAreaField
				fieldId={'ta1'}
				label={'Hi Roy'}
				description={'Say Hi'}
				value={'Roy'}
				onChange={onChange}
				onBlur={onBlur}
				attributes={{
					maxlength: 5, minlength: 1,cols:8,rows:5
				}}
			/>
		);
		expect(component).toMatchSnapshot();
	});

	it.skip('Passes rows and cols props', () => {
		const component = mount(
			<TextAreaField
				fieldId={'ta1'}
				label={'Hi Roy'}
				description={'Say Hi'}
				placeholder={'Hello'}
				html5type={'email'}
				value={'Roy'}
				onChange={onChange}
				onBlur={onBlur}
				attributes={{
					cols:8,rows:5
				}}
			/>
		);

		component.find('textarea').simulate('change');
		expect(component.find('textarea').props().cols ).toBe(8);
		expect(component.find('textarea').props().rows ).toBe(5);
	});


	it.skip('Changes calls change handler', () => {
		const component = mount(
			<TextAreaField
				fieldId={'ta1'}
				label={'Hi Roy'}
				description={'Say Hi'}
				placeholder={'Hello'}
				html5type={'email'}
				value={'Roy'}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);

		component.find('textarea').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
});
