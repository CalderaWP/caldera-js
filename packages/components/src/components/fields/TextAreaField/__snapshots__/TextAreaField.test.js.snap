// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaField  matches snapshot with all props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="ta1"
        >
          Hi Roy
           
        </label>
        <textarea
          class="form-control"
          cols="8"
          id="ta1"
          maxlength="5"
          minlength="1"
          rows="5"
        />
        <small
          class="text-muted form-text"
        >
          Say Hi
        </small>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="ta1"
      >
        Hi Roy
         
      </label>
      <textarea
        class="form-control"
        cols="8"
        id="ta1"
        maxlength="5"
        minlength="1"
        rows="5"
      />
      <small
        class="text-muted form-text"
      >
        Say Hi
      </small>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
